packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,205 @@
+          </section>
+          <section>
+            <Controller
+              name="cancellationDate"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <DatePicker
+                    id="cancellationDate"
+                    minDate={dayjs().startOf('day')}
+                    dateFormat="d/m/Y"
+                    datePickerType="single"
+                    value={value}
+                    onChange={([date]) => onChange(date)}
+                    autocomplete="off"
+                  >
+                    <DatePickerInput
+                      id="date-picker-calendar-id"
+                      placeholder="dd/mm/yyyy"
+                      labelText={t('cancellationDate', 'Cancellation date')}
+                      type="text"
+                      invalid={!!errors['cancellationDate']}
+                      invalidText={!!errors['cancellationDate'] && errors['cancellationDate'].message}
+                    />
+                  </DatePicker>
+                </div>
+              )}
+            />
+          </section>
+          <section>
+            <Controller
+              name="reasonForCancellation"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <TextArea
+                    type="text"
+                    id="reasonForCancellation"
+                    labelText={t('reasonForCancellation', 'Reason for cancellation')}
+                    value={value}
+                    onChange={(evt) => onChange(evt.target.value)}
+                    invalid={!!errors['reasonForCancellation']}
+                    invalidText={!!errors['reasonForCancellation'] && errors['reasonForCancellation'].message}
+                  />
+                </div>
+              )}
+            />
+          </section>
+        </Stack>
+      </div>
+
+      {showErrorNotification && (
+        <Column className={styles.errorContainer}>
+          <InlineNotification
+            lowContrast
+            title={t('error', 'Error')}
+            subtitle={t('pleaseFillRequiredFields', 'Please fill all the required fields') + '.'}
+            onClose={() => setShowErrorNotification(false)}
+          />
+        </Column>
+      )}
+
+      <ButtonSet className={isTablet ? styles.tablet : styles.desktop}>
+        <Button className={styles.button} kind="secondary" onClick={() => closeWorkspace()}>
+          {t('discard', 'Discard')}
+        </Button>
+        <Button
+          className={styles.button}
+          kind="primary"
+          onClick={handleSubmit(cancelOrderRequest, onError)}
+          disabled={isSubmitting}
+          type="submit"
+        >
+          {isSubmitting ? (
+            <InlineLoading description={t('saving', 'Saving') + '...'} />
```

Comment:
You'll need to add styling similar to [this](https://github.com/openmrs/openmrs-esm-core/blob/ca0c93abed7077af777e822d10c180feca0b02f9/packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-modal.scss#L17) for the loading spinner.

Reply:
Already done